state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1368,7 +1570,10 @@ impl<T: Config> SwapQueueApi for Pallet<T> {
 
 		let execute_at = frame_system::Pallet::<T>::block_number() + SWAP_DELAY_BLOCKS.into();
 
-		SwapQueue::<T>::append(execute_at, Swap::new(swap_id, from, to, swap_amount, swap_type));
+		SwapQueue::<T>::append(
```

Comment:
Should we check here that the execute_at block isn't later than the refund block? (I know it's unlikely)

Or alternatively we could do the expiry checks *before* executing the batch instead of after.

Reply:
Not only is it unlikely, it is actually impossible since it is only called on ingress, where refund_block is calculated at well from the current block. Maybe we could make this more clear somehow, but I think swapping redesign could potentially address this too.